submissions/andrewklmn/friends-app/app.js

```diff
@@ -0,0 +1,358 @@
+    resetButton.classList.remove('off');
+    return;
+  };
+  resetButton.classList.toggle('off',true);
+}
+
+const resetButtonClickHandler = ({filters, sorter}) => {
+
+  if (resetButton.classList.contains('off')) {
+    return;
+  }
+
+  const change = new Event('change');
+
+  filters.partOfName = searchField.value = '';
+  filters.gender = filterGender.value = '';
+  filters.ageRange[0] = filterMinAge.value = minPersonAge;
+  filters.ageRange[1] = filterMaxAge.value = maxPersonAge;
+  filters.country = filterCountry.value = '';
+
+  sorter.keyName = sortField.value = '';
+  sortField.dispatchEvent(change);
+}
+
+const redrawFriends = (state) => {
+  const {filters, sorter, initialListLength} = state;
+  preloader.classList.remove('hidden');
+  setResetButtonView(filters, sorter);
+  state.numberOfShowedFriends = 0;
+  container.innerHTML = '';
+  container.scrollTop = 0;
+
+  sortList(filterList(state))
+    .slice(0, initialListLength)
+    .forEach((friend) => {
+      state.numberOfShowedFriends++;
+      drawPerson(friend);
+    });
+
+  preloader.classList.add('hidden');
+}
+
+const updateFriendsList = (list)=> {
+  state.friends = list;
+  redrawFriends(state);
+}
+
+const addOptionToSelect = (select, optionValue, optionText) => {
+  const option = document.createElement('option');
+  option.value = optionValue;
+  option.innerHTML = optionText;
+  select.appendChild(option);
+}
+
+const drawSorter = (state)=>{
+  const {sorter} = state;
+  Object.keys(sorter.fieldPathList).forEach((field)=>{
+    addOptionToSelect(sortField, field, field);
+  });
+}
+
+const drawFilters = (state) => {
+
+  const onlyUnique = (value, index, self) => {
+    return self.indexOf(value) === index;
+  };
+  const {friends, filters} = state;
+
+  filters.genderList = friends.map(person => person.gender).filter(onlyUnique);
+  filters.genderList.forEach((gender)=>{
+    addOptionToSelect(filterGender, gender, gender);
+  });
+  filterMinAge.value = filters.ageRange[0];
+  filterMaxAge.value = filters.ageRange[1];
+  filters.countryList = friends.map(person => person.location.country).filter(onlyUnique);
+  filters.countryList.sort().forEach( country => {
+    addOptionToSelect(filterCountry, country, country);
+  });
+};
+
+const initApp = (state) => {
+
+  fetch(randomUserUrl)
+    .then(response => response.json())
+    .then(({results}) => {
+      updateFriendsList(results);
+      drawSorter(state);
+      drawFilters(state);
+      preloader.classList.add('hidden');
+    })
+    .catch(function() {
+      const header = document.querySelector('.header');
+      const footer = document.querySelector('.footer')
+      header.innerHTML = "ERROR: Can't get friend list!";
+      header.classList.add('error');
+      footer.innerHTML = "Try Refresh this page later!";
+      footer.classList.add('error');
+    });
+};
+
+const drawMoreFriends = (friends) => {
+  
+  friends.forEach((friend) => {
+    state.numberOfShowedFriends++;
```

Comment:
Maybe it's better to increase this variable by one operation:
`state.numberOfShowedFriends += friends.length`
```suggestion
    friends.forEach(friend => drawPerson(friend));
    state.numberOfShowedFriends += friends.length
```